### PR TITLE
initial commit of gp optimizer

### PR DESCRIPTION
**Pull request by thomaswmorris**
This is a (functional) draft, it needs quite a bit more work

### CODE REVIEW

**Review by AbbyGi — APPROVED** [review]

File: bloptools/gp/kernels.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+class LatentMaternKernel(gpytorch.kernels.Kernel):
+
+    def __init__(
+        self,
+        n_dof,
+        length_scale_bounds,
+        off_diag=True,
+        **kwargs,
+    ):
+        super(LatentMaternKernel, self).__init__(**kwargs)
```
Comment (mrakitin) [inline bloptools/gp/kernels.py:17]: I don't think the arguments are _required_ for the `super()` call. I think it was needed in the past with Py2.x.

File: bloptools/gp/plans.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        start_time = ttime.monotonic()
+        while (ttime.monotonic() - start_time < timeout) and (self.shutter.status.get() != 1):
+            print(f'Shutter not closed, retrying ... (closed_status = {self.shutter.status.get()})')
+            yield from bps.sleep(1.0)
+            yield from bps.mv(self.shutter.close_cmd, 1)
+
+        yield from bp.count([self.detector])
+
+        yield from bps.mv(self.shutter.open_cmd, 1)
+        yield from bps.sleep(2.0)
```
Comment (mrakitin) [inline bloptools/gp/plans.py:37]: Let's take a note that this code should be handled on the IPython startup files' side.

**Review by mrakitin — COMMENTED** [review]
I think it looks good overall! We need to work on extracting the shutter logic from here to the `profile_collection`. We can look at this tomorrow.

**Review by thomaswmorris — COMMENTED** [review]
LGTM!

**Review by mrakitin — APPROVED** [review]
I think it's good to go in. The shutter logic can be fixed later.

**Pull request merged**